test/dotnet-test.Tests/AssemblyInfo.cs

```diff
@@ -0,0 +1,4 @@
+using Xunit;
+
+// Don't let the tests execute concurrently
+[assembly: CollectionBehavior(CollectionBehavior.CollectionPerAssembly)]
```

Comment:
That said, I haven't seen the issue crop up since adding this, so perhaps it's not super terrible?